bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/handler/RemoteopenhabBridgeHandler.java

```diff
@@ -0,0 +1,554 @@
+
+    private void startCheckConnectionJob(RemoteopenhabRestClient client) {
+        ScheduledFuture<?> localCheckConnectionJob = checkConnectionJob;
+        if (localCheckConnectionJob == null || localCheckConnectionJob.isCancelled()) {
+            checkConnectionJob = scheduler.scheduleWithFixedDelay(() -> {
+                long millisSinceLastEvent = System.currentTimeMillis() - client.getLastEventTimestamp();
+                if (millisSinceLastEvent > CONNECTION_TIMEOUT_MILLIS) {
+                    logger.debug("Check: Disconnected from streaming events, millisSinceLastEvent={}",
+                            millisSinceLastEvent);
+                    checkConnection(client);
+                } else {
+                    logger.debug("Check: Receiving streaming events, millisSinceLastEvent={}", millisSinceLastEvent);
+                }
+            }, 0, CONNECTION_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void stopCheckConnectionJob() {
+        ScheduledFuture<?> localCheckConnectionJob = checkConnectionJob;
+        if (localCheckConnectionJob != null) {
+            localCheckConnectionJob.cancel(true);
+            checkConnectionJob = null;
+        }
+    }
+
+    private void restartStreamingUpdates() {
+        RemoteopenhabRestClient client = restClient;
+        if (client != null) {
+            synchronized (client) {
+                stopStreamingUpdates();
+                startStreamingUpdates();
+            }
+        }
+    }
+
+    private void startStreamingUpdates() {
+        RemoteopenhabRestClient client = restClient;
+        if (client != null) {
+            synchronized (client) {
+                client.addStreamingDataListener(this);
+                client.start();
+            }
+        }
+    }
+
+    private void stopStreamingUpdates() {
+        RemoteopenhabRestClient client = restClient;
+        if (client != null) {
+            synchronized (client) {
+                client.stop();
+                client.removeStreamingDataListener(this);
+            }
+        }
+    }
+
+    @Override
+    public void onConnected() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void onError(String message) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
+    }
+
+    @Override
+    public void onItemStateEvent(String itemName, String stateType, String state) {
+        updateChannelState(itemName, stateType, state);
+    }
+
+    @Override
+    public void onItemAdded(Item item) {
+        createChannels(List.of(item), false);
+    }
+
+    @Override
+    public void onItemRemoved(Item item) {
+        removeChannels(List.of(item));
+    }
+
+    @Override
+    public void onItemUpdated(Item newItem, Item oldItem) {
+        if (!newItem.type.equals(oldItem.type)) {
+            List<Item> items = new ArrayList<>();
+            items.add(newItem);
+            createChannels(items, false);
```

Comment:
```suggestion
            createChannels(List.of(newItem), false);
```

Reply:
I already changed 2 just below this morning but forgot this one.
Done